panel/reactive.py

```diff
@@ -4,38 +4,38 @@
 models rendered on the frontend.
 """
 
-import difflib
 import datetime as dt
+import difflib
 import logging
 import re
 import sys
 import textwrap
-
 from collections import Counter, defaultdict, namedtuple
 from functools import partial
+from pprint import pformat
+from typing import TYPE_CHECKING, Callable, Dict, List, Optional, Tuple, Union
 
 import bleach
 import numpy as np
 import param
-
 from bokeh.model import DataModel
 from param.parameterized import ParameterizedMetaclass, Watcher
-from pprint import pformat
 
 from .io.document import unlocked
 from .io.model import hold
 from .io.notebook import push
 from .io.state import set_curdoc, state
-from .models.reactive_html import (
-    ReactiveHTML as _BkReactiveHTML, ReactiveHTMLParser
-)
+from .models.reactive_html import ReactiveHTML as _BkReactiveHTML
+from .models.reactive_html import ReactiveHTMLParser
 from .util import edit_readonly, escape, updating
 from .viewable import Layoutable, Renderable, Viewable
 
-log = logging.getLogger('panel.reactive')
+if TYPE_CHECKING:
+    import pandas as pd
 
-LinkWatcher = namedtuple("Watcher", Watcher._fields+('target', 'links', 'transformed', 'bidirectional_watcher'))
+log = logging.getLogger('panel.reactive')
 
+LinkWatcher: Tuple = namedtuple("Watcher", Watcher._fields+('target', 'links', 'transformed', 'bidirectional_watcher'))
```

Comment:
A `Tuple` is not specific enough. But `collections.namedtuple` is a factory function and not a class. So there is no meaningfull `namedtuple` class. Instead from Python 3.6 `typing.NamedTuple` is provided. See https://www.geeksforgeeks.org/typing-namedtuple-improved-namedtuples/. I'm not sure it would work here as you add attributes dynamically using `Watcher._fields`.